pkg/proxy/endpoints.go

```diff
@@ -17,16 +17,63 @@ limitations under the License.
 package proxy
 
 import (
+	"net"
 	"reflect"
+	"strconv"
 	"sync"
 
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/record"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	utilproxy "k8s.io/kubernetes/pkg/proxy/util"
 )
 
+// BaseEndpointInfo contains common endpoint information.
+type BaseEndpointInfo struct {
+	Endpoint string // TODO: should be an endpointString type
+	// IsLocal indicates whether the endpoint is running in same host as kube-proxy.
+	IsLocal bool
+}
+
+var _ Endpoint = &BaseEndpointInfo{}
+
+// String is part of proxy.Endpoint interface.
+func (info *BaseEndpointInfo) String() string {
+	return info.Endpoint
+}
+
+// IsLocal is part of proxy.Endpoint interface.
+func (info *BaseEndpointInfo) GetIsLocal() bool {
```

Comment:
comment does not match code - IsLocal vs. GetIsLocal

I am curious why you changed it to GetIsLocal, I think the original name is good enough.

Reply:
Thanks, will fix, echo my comment above: https://github.com/kubernetes/kubernetes/pull/56880#discussion_r170387553. Basically it can not compile.

Reply:
got it. Thanks for clarifying :)